### DISCUSSION TITLE
JDTLS and Maven are not pointing to same maven repositories.

### DISCUSSION THREAD

**Discussion by kishooore** [mfussenegger/nvim-jdtls#343] [Q&A] [answered]
After configuring JDTLS, if I open any java file in the maven project, JDTLS downloads all the project dependencies even though maven has already downloaded all the dependencies into the .m2 folder. It is not reusing the .m2 folder dependencies.

Please help me if I am missing any configurations.

**Comment by mfussenegger** [comment, marked as answer]
Afaik there's no option to change that. But you'd need to ask in https://github.com/eclipse/eclipse.jdt.ls to be sure

**Reply by kishooore** [reply]
Thank you for your response.

**Comment by kareem-abdul** [comment]
This can be configured, you can add your maven settings file inside your jdtls config like so
```lua
                configuration = {
                    runtimes = {
                        ...
                    },
                    maven = {
                        userSettings =  '/path/to/maven/settings.xml'
                    }
                }
```
and provide the local repo folder inside the settings.xml file like so
```xml
 <localRepository>${user.home}/.local/lib/maven</localRepository>
```
https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2696#issuecomment-1586328582
